pkg/cmd/cli/cmd/login/loginoptions.go

```diff
@@ -92,75 +97,77 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 		}
 	}
 
-	// normalize the provided server to a format expected by config
-	serverNormalized, err := config.NormalizeServerURL(o.Server)
-	if err != nil {
+	if err := o.normalizeServerURL(); err != nil {
 		return nil, err
 	}
-	o.Server = serverNormalized
-	clientConfig.Host = o.Server
-
-	if len(o.CAFile) > 0 {
-		clientConfig.CAFile = o.CAFile
 
-	} else {
-		// check all cluster stanzas to see if we already have one with this URL that contains a client cert
-		for _, cluster := range o.StartingKubeConfig.Clusters {
-			if cluster.Server == clientConfig.Host {
-				if len(cluster.CertificateAuthority) > 0 {
-					clientConfig.CAFile = cluster.CertificateAuthority
-					break
-				}
-
-				if len(cluster.CertificateAuthorityData) > 0 {
-					clientConfig.CAData = cluster.CertificateAuthorityData
-					break
-				}
-			}
-		}
-	}
-
-	// ping to check if server is reachable
-	osClient, err := client.New(clientConfig)
-	if err != nil {
-		return nil, err
-	}
+	if err := o.dialToServer(); err != nil {
+		clientConfig.Host = o.Server
```

Comment:
Right, this was a leftover from when we was doing retries on other ports. Fixed.